test_convert_fastaqual_fastq.py permissions issue

`test_convert_fastaqual_fastq.py` fails `test_fastaqual_output` and `test_fastq_output` if the tests are run as `root`. This issue was brought up in [this QIIME forum post](https://groups.google.com/d/topic/qiime-forum/cgnckA9StWY/discussion).

Here is the test failure output:

```
.....F....F.....
======================================================================
FAIL: test_fastaqual_output (__main__.MakeFastaqualTests)
Raises error if cannot open output filepath
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_convert_fastaqual_fastq.py", line 278, in test_fastaqual_output
    output_directory = self.read_only_output_dir)
AssertionError: IOError not raised

======================================================================
FAIL: test_fastq_output (__main__.MakeFastqTests)
Raises errors when can't write output file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_convert_fastaqual_fastq.py", line 149, in test_fastq_output
    self.qual_file_path, output_directory = self.read_only_output_dir )
AssertionError: IOError not raised

----------------------------------------------------------------------
Ran 16 tests in 0.184s

FAILED (failures=2)
```

I think the issue has to do with the following lines in the tests:

```
self.read_only_output_dir = get_tmp_filename(prefix = 'read_only_', \
suffix = '/')
create_dir(self.read_only_output_dir)
# Need read only directory to test errors for files written during
# fastq/fasta iteration.
chmod(self.read_only_output_dir, 0577)
```

I think changing the permissions to `0555` will fix this issue.

--- a/tests/test_convert_fastaqual_fastq.py
+++ b/tests/test_convert_fastaqual_fastq.py
@@ -2,7 +2,7 @@
 
 __author__ = "Adam Robbins-Pianka, Abhisaar Yadav"
 __copyright__ = "Copyright 2011, The QIIME project"
-__credits__ = ["Adam Robbins-Pianka, Abhisaar Yadav"]
+__credits__ = ["Adam Robbins-Pianka", "Abhisaar Yadav", "Jai Ram Rideout"]
 __license__ = "GPL"
 __version__ = "1.6.0-dev"
 __maintainer__ = "Adam Robbins-Pianka"
@@ -58,7 +58,7 @@ class MakeFastqTests(TestCase):
         create_dir(self.read_only_output_dir)
         # Need read only directory to test errors for files written during
         # fastq/fasta iteration.
-        chmod(self.read_only_output_dir, 0577)
+        chmod(self.read_only_output_dir, 0555)
 
         self.output_dir = get_tmp_filename(prefix = 'convert_fastaqual_fastq_',\
                                            suffix = '/')
@@ -196,7 +196,7 @@ class MakeFastaqualTests(TestCase):
         self.read_only_output_dir = get_tmp_filename(prefix = 'read_only_', \
         suffix = '/')
         create_dir(self.read_only_output_dir)
-        chmod(self.read_only_output_dir, 0577)
+        chmod(self.read_only_output_dir, 0555)
 
         self.output_dir = get_tmp_filename(prefix = 'convert_fastaqual_fastq_',\
          suffix = '/')
@@ -328,7 +328,7 @@ class ConvertFastaqualTests(TestCase):
         self.read_only_output_dir = get_tmp_filename(prefix = 'read_only_', \
         suffix = '/')
         create_dir(self.read_only_output_dir)
-        chmod(self.read_only_output_dir, 0577)
+        chmod(self.read_only_output_dir, 0555)
 
         self.output_dir = get_tmp_filename(prefix = 'convert_fastaqual_fastq_',\
          suffix = '/')

